cirq/google/calibration/workflow.py

```diff
@@ -337,19 +351,150 @@ def run_characterizations(
     return results
 
 
-@dataclasses.dataclass(frozen=True)
-class CircuitPhasedFSimCalibrationResults:
-    """Circuit-specific calibration results
+def zeta_chi_gamma_calibration_for_moments(
+    circuit_calibration: CircuitCalibration,
+    characterizations: List[PhasedFSimCalibrationResult],
+    gates_translator: Callable[[Gate], Optional[FSimGate]] = try_convert_sqrt_iswap_to_fsim,
+) -> CircuitCalibration:
+    """Compensates circuit against errors in zeta, chi and gamma angles.
+
+    This method creates a new circuit with a single-qubit Z gates added in a such way so that
+    zeta, chi and gamma angles discovered by characterizations are cancelled-out and set to 0.
+
+    This function preserves a moment structure of the circuit. All single qubit gates appear on new
+    moments in the final circuit.
+
+    Args:
+        circuit_calibration: Description of the circuit together with its calibration metadata that
+            is compatible with the characterizations argument.
+        characterizations: List of characterization results.
+        gates_translator: Function that translates a gate to a supported FSimGate which will undergo
+            characterization. Defaults to sqrt_iswap_gates_translator.
+
+    Returns:
+        Calibrated circuit together with its calibration metadata in CircuitCalibration object. The
+        calibrated circuit has single-qubit Z gates added which compensates for the true gates
+        imperfections.
+    """
+    if len(circuit_calibration.circuit) != len(circuit_calibration.moment_allocations):
+        raise ValueError('Moment allocations does not match circuit length')
+
+    default_phases = PhasedFSimCharacterization(zeta=0.0, chi=0.0, gamma=0.0)
+
+    compensated = Circuit()
+    compensated_allocations = []
+    for moment, characterization_index in zip(
+        circuit_calibration.circuit, circuit_calibration.moment_allocations
+    ):
+        parameters = None
+        if characterization_index is not None:
+            parameters = characterizations[characterization_index]
+
+        decompositions = []
+        other = []
```

Comment:
type annotations could be helpful here for `decompositions` `other` and `new_moment_mapping`

Reply:
Done